spec/unit/resource/api200/login_detail_spec.rb

```diff
@@ -0,0 +1,10 @@
+require 'spec_helper'
+
+RSpec.describe OneviewSDK::LoginDetail do
+  include_context 'shared context'
+  describe '#get_login_details' do
+    it 'calls the uri' do
+      allow(@client_200).to receive(:rest_get).with(described_class::BASE_URI, {})
```

Comment:
This test is not testing the call of method `rest_get`, because the `allow` just permit you call some method, but it is not guaranteed to be called. So, a good way is do this: 
```ruby
    expect(@client_200).to receive(:rest_get).with(described_class::BASE_URI, {})
    described_class.get_login_details(@client_200)
```
because you are expecting that the calling to `described_class.get_login_details` execute `@client_200.rest_get`

Reply:
Also, rename your unit test files from login_detail.rb to login_detail_spec.rb for api300 and api500